### PR TITLE
chore(eslint): migrate .eslintrc to eslint.config.js

### PR DESCRIPTION
**Pull request by sukvvon**
## Summary

* mirgate legacy `.eslintrc` style to `eslint.config.js`

ref: https://github.com/pmndrs/zustand/pull/2912

## Check List

- [x] `pnpm run prettier` for formatting code and docs

### CODE REVIEW

**Comment by sukvvon** [comment]
image

@dai-shi to prevent warning i add `@typescript-eslint/no-unused-vars: 'off'`

**Comment by dai-shi** [comment]
> to prevent warning i add `@typescript-eslint/no-unused-vars: 'off'`

Please keep the rule and add `_` prefix.

**Comment by sukvvon** [comment]
> > to prevent warning i add `@typescript-eslint/no-unused-vars: 'off'`
> 
> Please keep the rule and add `_` prefix.

@dai-shi could you explain about `_` prefix?

**Comment by dai-shi** [comment]
For unused vars, we have a rule to exclude `^_` pattern, both for eslint and tsc.

File: .eslintrc.json [hunk 1]
```diff
@@ -1,121 +0,0 @@
...
-    {
-      "extends": [
-        "plugin:jest-dom/recommended",
-        "plugin:testing-library/react",
-        "plugin:@vitest/legacy-recommended"
-      ],
-      "files": ["tests/**/*.ts", "tests/**/*.tsx"],
-      "rules": {
-        "import/extensions": ["error", "never"],
-        "@typescript-eslint/no-unused-vars": "off",
```
Comment (dai-shi) [inline .eslintrc.json:106]: Oh, I see. We already have it here. Never mind then.

**Comment by dai-shi** [comment]
> to prevent warning i add `@typescript-eslint/no-unused-vars: 'off'`

So, it's not you who added it. It was there and you just migrated it.

**Comment by sukvvon** [comment]
> > to prevent warning i add `@typescript-eslint/no-unused-vars: 'off'`
> 
> So, it's not you who added it. It was there and you just migrated it.

@dai-shi i think i forgot haha it's my bad 😂

**Comment by sukvvon** [comment]
> For unused vars, we have a rule to exclude `^_` pattern, both for eslint and tsc.

@dai-shi right, i already migrate it too

**Comment by sukvvon** [comment]
@dai-shi Is there anything that i missed? 🤔

**Comment by dai-shi** [comment]
Just busy. Would you like to join my discord server? https://discord.gg/MrQdmzd

**Comment by sukvvon** [comment]
> Just busy. Would you like to join my discord server? https://discord.gg/MrQdmzd

I joined!

**Comment by sukvvon** [comment]
`pnpm-lock.yaml` is an auto-generated file, and manual edits can cause dependency mismatches or tree corruption, so a new PR is created.

**Pull request closed without merging**